js/client/modules/@arangodb/air/single-source-shortest-paths.js

```diff
@@ -46,121 +47,159 @@ exports.test = test;
 
 */
 function single_source_shortest_paths_program(
-  resultField,
-  startVertexId,
-  weightAttribute
+    resultField,
+    startVertexId,
+    weightAttribute
 ) {
-  return {
-    resultField: resultField,
-    maxGSS: 10000,
-    vertexAccumulators: {
-      distance: {
-        accumulatorType: "min",
-        valueType: "doubles",
-        storeSender: true,
-      },
-    },
-    phases: [
-      {
-        name: "main",
-        initProgram: [
-          "seq",
-          [
-            "if",
-            [
-              ["eq?", ["this-vertex-id"], startVertexId],
-              ["seq", ["accum-set!", "distance", 0], true],
-            ],
-            [true, ["seq",
-                    ["accum-clear!", "distance"],
-                    false]],
-          ],
+    return {
+        resultField: resultField,
+        maxGSS: 10000,
+        vertexAccumulators: {
+            distance: {
+                accumulatorType: "custom",
+                valueType: "slice",
+                customType: "my_min",
+                storeSender: true,
+            },
+        },
+        customAccumulators: {
+            "my_min": {
```

Comment:
It'd be nice if we had a `min-accumulator.js` file in our directory that contains a constant `min_accumulator` which is just this JSON that is the min-accumulator program, then we can refer to this min-accumulator in all algorithms already (like pedestrian libraries)